src/snowflake/connector/s3_util.py

```diff
@@ -71,13 +70,16 @@ def create_client(stage_info, use_accelerate_endpoint=False):
             s3={
                 'use_accelerate_endpoint': use_accelerate_endpoint,
                 'addressing_style': ADDRESSING_STYLE
-            })
-        client = boto3.resource(
-            's3',
+            }
+        )
+        session = boto3.Session(
```

Comment:
Did you verify this address the issue? Is the

Reply:
I thought it was not well documented based on the linked thread, but it looks like right here boto3 documents thread safety: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/resources.html#multithreading-and-multiprocessing
Let me reach out to the PR's author

Reply:
Ok, the doc looks good enough. Thanks for checking.